tests/manage/test_create_storage_class_pvc.py

```diff
@@ -1,135 +1,127 @@
+    RBD_SC_OBJ = helpers.create_storage_class(
+        interface_type=constants.CEPHBLOCKPOOL,
+        interface_name=RBD_POOL.name,
+        secret_name=RBD_SECRET_OBJ.name
+    )
 
 
-def setup_fs(self):
+def teardown_rbd():
     """
-    Setting up the environment for the test
+    Tearing down the environment
+    Deleting pod,replicated pool,pvc,storageclass,secret of rbd
     """
-    global CEPH_OBJ
-    self.fs_data = templating.load_yaml_to_dict(constants.CEPHFILESYSTEM_YAML)
-    self.fs_data['metadata']['name'] = helpers.create_unique_resource_name(
-        'test', 'cephfs'
-    )
-    self.fs_data['metadata']['namespace'] = config.ENV_DATA['cluster_namespace']
-    CEPH_OBJ = OCS(**self.fs_data)
-    CEPH_OBJ.create()
-    assert POD.wait_for_resource(
-        condition='Running', selector='app=rook-ceph-mds'
+    global RBD_PVC_OBJ, RBD_POD_OBJ
+    log.info('deleting rbd pod')
+    RBD_POD_OBJ.delete()
+    log.info("Deleting RBD PVC")
+    RBD_PVC_OBJ.delete()
+    assert helpers.validate_pv_delete(RBD_SC_OBJ.name)
+    log.info("Deleting CEPH BLOCK POOL")
+    RBD_POOL.delete()
+    log.info("Deleting RBD Secret")
+    RBD_SECRET_OBJ.delete()
+    log.info("Deleting RBD Storageclass")
+    RBD_SC_OBJ.delete()
+    log.info("Deleting CephFS PVC")
+
+
+@pytest.fixture(scope='function')
+def test_fixture_cephfs(request):
+    def finalizer():
+        teardown_fs()
+
+    request.addfinalizer(finalizer)
+    setup_fs()
+
+
+def setup_fs():
+    log.info("Creating CEPHFS Secret")
+    global CEPHFS_SECRET_OBJ
+    CEPHFS_SECRET_OBJ = helpers.create_secret(constants.CEPHFILESYSTEM)
+
+    global CEPHFS_SC_OBJ
+    log.info("Creating CephFS Storage class ")
+    CEPHFS_SC_OBJ = helpers.create_storage_class(
+        constants.CEPHFILESYSTEM,
+        helpers.get_cephfs_data_pool_name(),
+        CEPHFS_SECRET_OBJ.name
     )
-    pods = POD.get(selector='app=rook-ceph-mds')['items']
-    assert len(pods) == 2
 
 
 def teardown_fs():
-    """
-    Tearing down the environment
-    """
-    global CEPH_OBJ
-    CEPH_OBJ.delete()
+    global CEPHFS_PVC_OBJ, CEPHFS_POD_OBJ
+    log.info('deleting cephfs pod')
+    CEPHFS_POD_OBJ.delete()
+    log.info('deleting cephfs pvc')
+    CEPHFS_PVC_OBJ.delete()
+    assert helpers.validate_pv_delete(CEPHFS_SC_OBJ.name)
+    log.info("Deleting CEPHFS Secret")
+    CEPHFS_SECRET_OBJ.delete()
+    log.info("Deleting CephFS Storageclass")
+    CEPHFS_SC_OBJ.delete()
 
 
 @tier1
 class TestOSCBasics(ManageTest):
-    mons = (
-        f'rook-ceph-mon-a.{config.ENV_DATA["cluster_namespace"]}'
-        f'.svc.cluster.local:6789,'
-        f'rook-ceph-mon-b.{config.ENV_DATA["cluster_namespace"]}.'
-        f'svc.cluster.local:6789,'
-        f'rook-ceph-mon-c.{config.ENV_DATA["cluster_namespace"]}'
-        f'.svc.cluster.local:6789'
-    )
-
     @pytest.mark.polarion_id("OCS-336")
-    def test_basics_rbd(self, test_fixture):
+    def test_basics_rbd(self, test_fixture_rbd):
```

Comment:
this will not work as you are not using yield in the fixture
@red-hat-storage/top-level-reviewers , please correct me if I'm wrong

Reply:
He is using the alternate finalizer method here so should work.